wipp-backend-data/src/main/java/gov/nist/itl/ssd/wipp/backend/data/csvCollection/csv/CsvUploadController.java

```diff
@@ -0,0 +1,225 @@
+                    + " does not exist.");
+        }
+//        if (oldTc.get().isLocked()) {
+//            throw new ClientException("Collection locked.");
+//        }
+    }
+
+    @Override
+    protected FlowFile getFlowFile(HttpServletRequest request,
+                                   Parameters parameters) {
+        FlowFile flowFile = super.getFlowFile(request, parameters);
+        return new CsvCollectionFlowFile(flowFile,
+                (CsvCollectionFileParameters) parameters);
+    }
+
+    @Override
+    protected File getUploadDir(FlowFile flowFile) {
+        return getUploadDir(getCollectionId(flowFile));
+    }
+
+    protected File getUploadDir(String csvCollectionId) {
+        return new File(config.getCsvCollectionsFolder(), csvCollectionId);
+    }
+
+    @Override
+    protected File getTempUploadDir(FlowFile flowFile) {
+        return getTempUploadDir(getCollectionId(flowFile));
+    }
+
+    protected File getTempUploadDir(String csvCollectionId) {
+        return new File(config.getCsvCollectionsUploadTmpFolder(), csvCollectionId);
+    }
+
+    @Override
+    protected void onUploadFinished(FlowFile flowFile, Path tempPath)
+            throws IOException {
+        String collectionId = getCollectionId(flowFile);
+        String fileName = flowFile.getFlowFilename();
+
+        try {
+            CsvCollection csvCol = csvCollectionRepository.findById(collectionId).get();
+        } catch (NoSuchElementException e) {
+            LOG.log(Level.WARNING, "Error finding collection " + collectionId
+                            + " when uploading file " + fileName,
+                    e);
+        }
+
+        if(fileName != null){
+            fileName = fileName.replaceAll("[\\p{Punct}&&[^.-]]", "_");
+            fileName = fileName.replace(" ", "");
+            uploadCsv(flowFile, tempPath, fileName);
+        }
+    }
+
+    private void uploadCsv(FlowFile flowFile, Path tempPath, String fileName) throws IOException{
+        File uploadDir = getUploadDir(flowFile);
+        uploadDir.mkdirs();
+        String collectionId = getCollectionId(flowFile);
+        Csv csv = new Csv(collectionId, fileName, flowFile.getFlowFilename(),
+                getPathSize(tempPath), true);
+        csvRepository.save(csv);
+        csvCollectionRepository.updateCsvCaches(collectionId);
+        Path outputPath = new File(uploadDir, fileName).toPath();
+        Files.copy(tempPath, outputPath);
+        Files.delete(tempPath);
+        csv.setFileName(fileName);
+        csv.setFileSize(getPathSize(outputPath));
+        csv.setImporting(false);
+        csvRepository.save(csv);
+        csvCollectionRepository.updateCsvCaches(collectionId);
+    }
+
+    protected static String getCollectionId(FlowFile flowFile) {
+        CsvCollectionFlowFile tff = (CsvCollectionFlowFile) flowFile;
+        return tff.collection;
+    }
+
+    protected static long getPathSize(Path path) {
```

Comment:
With the latest changes made for images collections, this method is already in `FlowJsController`, so you should not need it here